apps/prairielearn/src/batched-migrations/20250214035153_course_instance_usages__submissions__backfill.ts

```diff
@@ -0,0 +1,41 @@
+import { z } from 'zod';
+
+import { makeBatchedMigration } from '@prairielearn/migrations';
+import { loadSqlEquiv, queryAsync, queryRow } from '@prairielearn/postgres';
+
+import { updateCourseInstanceUsagesForSubmission } from '../models/course-instance-usages.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+const CUTOFF_DATE = '2025-02-15T00:00:00Z';
+
+export default makeBatchedMigration({
+  async getParameters() {
+    // First delete usage data older than a hard-coded date
+    await queryAsync(sql.delete_old_usages, { CUTOFF_DATE });
```

Comment:
Nit: put this in `apps/prairielearn/src/migrations/20250214035153_course_instance_usages__submissions__backfill.ts` instead? It feels cleaner to me to say "delete all the data and then start the batched migration", even though I know the end result is the same because I know how batched migrations work 🙂

Reply:
I don't disagree that this feels a bit out of place, but I had two issues with doing it in the migration:

1. The migration is a TS file, so to run SQL from it I would ideally have a file with the same name but ending in `.sql`. I believe that doing this will be seen as a duplicate timestamp, however. Storing the SQL elsewhere feels nonstandard.
2. We are passing a single `CUTOFF_DATE` constant to both the delete query and the backfill, to make sure we are being consistent. Shifting the delete to the migration means we'll need to import this variable from the batched migration, which I guess is fine but also feels like extra complexity.

Thoughts?